File: SDKs/zuora.com/2021-08-20/typescript/src/sdk/models/shared/putorderpatchrequesttype.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



export class PutOrderPatchRequestTypeSubscriptionsOrderActions extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=customFields" })
  customFields?: Record<string, any>;

  @SpeakeasyMetadata({ data: "json, name=orderActionId" })
  orderActionId?: string;

  @SpeakeasyMetadata({ data: "json, name=sequence" })
  sequence?: number;
}


export class PutOrderPatchRequestTypeSubscriptions extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=orderActions", elemType: PutOrderPatchRequestTypeSubscriptionsOrderActions })
  orderActions?: PutOrderPatchRequestTypeSubscriptionsOrderActions[];

  @SpeakeasyMetadata({ data: "json, name=subscriptionNumber" })
  subscriptionNumber?: string;
}


export class PutOrderPatchRequestType extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=customFields" })
  customFields?: Record<string, any>;

  @SpeakeasyMetadata({ data: "json, name=subscriptions", elemType: PutOrderPatchRequestTypeSubscriptions })
  subscriptions?: PutOrderPatchRequestTypeSubscriptions[];
}

File: SDKs/zuora.com/2021-08-20/python/src/sdk/models/shared/putorderpatchrequesttype.py
import dataclasses
from typing import Any,Optional
from dataclasses_json import dataclass_json
from sdk import utils


@dataclass_json
@dataclasses.dataclass
class PutOrderPatchRequestTypeSubscriptionsOrderActions:
    custom_fields: Optional[dict[str, Any]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('customFields') }})
    order_action_id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('orderActionId') }})
    sequence: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('sequence') }})
    

@dataclass_json
@dataclasses.dataclass
class PutOrderPatchRequestTypeSubscriptions:
    order_actions: Optional[list[PutOrderPatchRequestTypeSubscriptionsOrderActions]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('orderActions') }})
    subscription_number: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('subscriptionNumber') }})
    

@dataclass_json
@dataclasses.dataclass
class PutOrderPatchRequestType:
    custom_fields: Optional[dict[str, Any]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('customFields') }})
    subscriptions: Optional[list[PutOrderPatchRequestTypeSubscriptions]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('subscriptions') }})
    
